lib/elements/mixer.ex

```diff
@@ -101,25 +92,25 @@ defmodule Basic.Elements.Mixer do
     end
   end
 
-  defp send_end_of_stream(state) do
+  defp maybe_send_end_of_stream(state) do
     end_of_stream_actions =
       if Enum.all?(state.tracks, fn {_, track} -> track.status == :finished end) do
         [end_of_stream: :output]
       else
         []
       end
-
     {state, end_of_stream_actions}
   end
 
-  defp demand_on_empty_tracks(state, pads) do
+  defp get_demand_actions(state, pads) do
```

Comment:
I am not sure about that name - not only are we getting demand actions, we are also returning state. 
Since we are not updating `state`, we can return only the `actions` and then the name would be ok (I believe the reason why the `{state, actions` tuple was originally returned was to make it possible to call that function at the end of some Membrane's callback.

Reply:
This name is used in the tutorial on the website, so when i was going through it i just copied it. I agree with your reasoning and I believe this function shouldn't return the state since it's not being changed

Reply:
Sure, I know that it's not you who have called this function that way, but I think we can change it since we are already refining this tutorial ;)